lib/formats/odata.js

```diff
@@ -350,10 +350,17 @@ const rowStreamToOData = (fields, table, domain, originalUrl, query, inStream, t
       // per row, we do our asynchronous parsing, jam the result onto the
       // text resultstream, and call done to indicate that the row is processed.
       submissionToOData(fields, table, row, options).then((data) => {
+
+        const parentIdProperty = Object.keys(data[0] ?? true).filter(p => /^__.*-id$/.test(p))[0];
+
         for (const field of data) {
+
+          // if ID (of subtable) is not requested then remove it
+          const fieldRefined = (options.metadata && !options.metadata.__id) ? (({ [parentIdProperty]: __, ...item }) => item)(field) : field;
```

Comment:
There's a utility function in lib/util/util.js named `without()` that I think would be helpful here.